gen-ai-rag-vertex-ai-vector-search/ingestion.tf

```diff
@@ -0,0 +1,77 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Aspects to setup ingestion from Cloud Storage bucket to processing job.
+
+resource "google_storage_bucket" "ingest" {
+  name     = "ingest-${local.unique_str}"
+  location = var.region
+}
+
+
+resource "google_pubsub_topic" "ingest" {
+  name = "ingest-${local.unique_str}"
+
+  ingestion_data_source_settings {
+    cloud_storage {
+      bucket = google_storage_bucket.ingest.name
+    }
+  }
+}
+
+# Allow the Pub/Sub service account the ability to publish messages
+resource "google_project_iam_member" "pubsub" {
+  project = var.project_id
+  role    = "roles/pubsub.publisher"
+  member  = "serviceAccount:service-${data.google_project.default.project_id}@gcp-sa-pubsub.iam.gserviceaccount.com"
```

Comment:
This service account would need to be created first, right, before assigning the role ?

Reply:
This is the API service account, which should be created when the API is enabled.